src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -125,6 +130,59 @@ List<SubstationGeoData> getSubstations(Network network, Set<Country> countries)
         return new ArrayList<>(substationsGeoData.values());
     }
 
+    List<SubstationGeoData> getSubstations(Network network, List<String> substationsIds) {
+        String escapedIds = StringUtils.join(substationsIds.stream().map(LogUtils::sanitizeParam).collect(Collectors.toList()), ", ");
+        LOGGER.info("Loading substations geo data for substations with ids {} of network '{}'", escapedIds, network.getId());
+
+        StopWatch stopWatch = StopWatch.createStarted();
+
+        Map<String, SubstationGeoData> substationsGeoData = new HashMap<>();
+
+        Set<String> substationsToCalculate = new HashSet<>();
+
+        Map<String, SubstationGeoData> geoDataForComputation = new HashMap<>();
+        Map<String, Set<String>> neighboursBySubstationId = new HashMap<>();
+
+        prepareGeoDataForComputation(network, geoDataForComputation, neighboursBySubstationId, substationsToCalculate, substationsIds.stream().collect(Collectors.toSet()));
+
+        //Calculated data are added to geoDataForComputation
+        calculateMissingGeoData(network, neighboursBySubstationId, geoDataForComputation, substationsToCalculate);
+        calculateDefaultSubstationsGeoData(geoDataForComputation, neighboursBySubstationId);
+
+        //We remove linked substations from result - we only want requested ones
+        geoDataForComputation.keySet().removeIf(key -> !substationsIds.contains(key));
```

Comment:
Why calculate some MissingGeoData or DefaultGeoDAta to remove it just after ?

Reply:
calculateMissingGeoData() is made such as substationsToCalculate are calculated and put in geoDataForComputation.

So I need to clean the result of computation to send back only geodata that were requested.

Reply:
Ok will see later If something must be refactor or not